packages/next/server/next-server.ts

```diff
@@ -2165,7 +2179,14 @@ export default class Server {
     }
 
     try {
-      await serveStatic(req, res, path)
+      const { hotReloader } = this as any as DevServer
```

Comment:
Can you make a `protected async getStaticFs(req)` or such and move the special case to `next-dev-server`?

Reply:
This is the only place where the `req` object is used though. Is there another place that could utilize the request?

Reply:
I added `getClientFilesystem()` in beb9d1c so we'll see if this works as expected